src/engine/IndexScan.h

```diff
@@ -49,7 +49,8 @@ class IndexScan final : public Operation {
             const TripleComponent& s, const TripleComponent& p,
             const TripleComponent& o,
             std::vector<ColumnIndex> additionalColumns,
-            std::vector<Variable> additionalVariables, Graphs graphsToFilter,
+            std::vector<Variable> additionalVariables,
+            const size_t numVariables, Graphs graphsToFilter,
```

Comment:
As the `numVariables` is dependent on the `s, p and o` it is cleaner to automatically compute it in the constructor (implement a free function for this).